components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -81,15 +74,73 @@ export default function () {
                         </a>
                     </span>
                 }
+                disabled={!!project.settings?.allowUsingPreviousPrebuilds}
                 checked={project.settings?.useIncrementalPrebuilds ?? false}
                 onChange={({ target }) => updateProjectSettings({ useIncrementalPrebuilds: target.checked })}
             />
             <CheckBox
                 title={<span>Cancel Prebuilds on Outdated Commits </span>}
                 desc={<span>Cancel pending or running prebuilds on the same branch when new commits are pushed.</span>}
                 checked={!project.settings?.keepOutdatedPrebuildsRunning}
+                disabled={!!project.settings?.allowUsingPreviousPrebuilds}
                 onChange={({ target }) => updateProjectSettings({ keepOutdatedPrebuildsRunning: !target.checked })}
             />
+            <h3 className="mt-12">Incremental Workspaces</h3>
+            <CheckBox
+                title={
+                    <span>
+                        Enable Incremental Workspaces{" "}
+                        <PillLabel type="warn" className="font-semibold mt-2 ml-2 py-0.5 px-2 self-center">
+                            Experimental
+                        </PillLabel>
+                    </span>
+                }
+                desc={
+                    <span>
+                        Use an earlier successful prebuild as a base to create new workspaces. This can speed up the
+                        workspace initialization drastically depending on how good your build (the init tasks) make use
+                        of the file system caches.
+                    </span>
+                }
+                checked={!!project.settings?.allowUsingPreviousPrebuilds}
+                onChange={({ target }) => updateProjectSettings({ allowUsingPreviousPrebuilds: target.checked })}
+            />
+            <div className="flex mt-4 max-w-2xl">
+                <div className="flex flex-col ml-6">
+                    <label
+                        htmlFor="prebuildNthCommit"
+                        className="text-gray-800 dark:text-gray-100 text-md font-semibold cursor-pointer tracking-wide"
+                    >
+                        Frequency of prebuilds
```

Comment:
I intentionally moved it here, because it only makes sense in this context. This is also the reason why I opted for a section called "Incremental Workspaces", i.e. to have them together rin one context.